packages/Interface/src/DTK_C_API.cpp

```diff
@@ -349,98 +363,135 @@ bool DTK_is_initialized() { return DataTransferKit::isInitialized(); }
 
 void DTK_finalize() { DataTransferKit::finalize(); }
 
-void DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
-                       void ( *f )(), void *user_data )
+int DTK_set_function( DTK_UserApplicationHandle handle, DTK_FunctionType type,
+                      void ( *f )(), void *user_data )
```

Comment:
Thinking further, if we always set `errno` to 0 for each of the interface functions, the user just need to
```c
 DTK_some_function(some_args);
if (errno)
  printf("DTK failed: %s\n", DTK_error(errno));
```
assuming we change `DTK_error` to match `strerror` prototype
```c
char * strerror (int errnum );
```